src/main/kotlin/com/sourcegraph/cody/history/state/LLMState.kt

```diff
@@ -13,12 +13,18 @@ class LLMState : BaseState() {
 
   @get:OptionTag(tag = "provider", nameAttribute = "") var provider: String? by string()
 
+  @get:OptionTag(tag = "tags", nameAttribute = "") var tags: MutableList<String> by list()
+
+  @get:OptionTag(tag = "usage", nameAttribute = "") var usage: MutableList<String> by list()
+
```

Comment:
I don't know the implications of this. Where is it getting this state from? Is this going to break immediately?

Reply:
This is state saved in a IntelliJ xml settings files.
You will need to add migration which will convert old format to the new one.
For the reference you can look at the `DeprecatedChatLlmMigration`.
 
In practice you will most likely have to:
1. Keep `ChatModelProvider::deprecated)` field but mark it as `@Deprecated`.
2. In the migration function (see `DeprecatedChatLlmMigration::migrate` for reference):
- read all the old chats
- for each chat read the content from `deprecated` field
- set `tags` appropriately
- reset `deprecated` to be `null` (as it won't be used anymore, we will remove it in future)

Reply:
Alright. I created `ChatTagsLllmMigration` that I believe accomplishes this.